library/ssl_tls.c

```diff
@@ -1973,6 +2021,10 @@ STATIC int ssl_decrypt_buf( mbedtls_ssl_context *ssl,
             return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
         }
 
+        /* If the padding was found to be invalid, padlen == 0
+         * and the subtraction is safe. If the padding was found valid,
+         * padlen hasn't been changed and the previous assertion
+         * data_len >= padlen still holds. */
```

Comment:
Just to clarify my review comment: this is one of the awkward references I'm talking about: we actually tested `data_len >= padlen + 1` but here we can only be sure that `data_len >= padlen` since we did `padlen+` in the meantime.

Reply:
Hmm, now that I understand you were referring to the comment, not the code, this particular point doesn't make sense any more.

However I still stand by the fact that I would prefer `padlen` to have the same meaning everywhere.